articles/debug-node-devtools/index.md

```diff
@@ -0,0 +1,113 @@
+
+A debugger is a program that is used to inspect a program as it runs. It enables running the code line-by-line and following the execution process. In the process, you can see the code as a flow chart and logically interpret it. This enables you to find bugs more easily.
+
+The most popular way JavaScript and Node developers track bugs in their programs is printing data to the console. They use methods like [console.log()](https://developer.mozilla.org/en-US/docs/Web/API/Console/log) and [console.debug()](https://developer.mozilla.org/en-US/docs/Web/API/Console/debug). Although these methods can be helpful, they require a lot of manual effort and are not foolproof. Suppose you have a large app, how many `console.log()` statements would you need? Hundreds, perhaps. 
+
+Logging sensitive information can occur, exposing your app to malicious exploitations and data breaches. Debuggers come handy in a situation like this. They help track bugs without exposing your app to security threats.
+
+Sometimes Node.js apps can fail without error messages. This occurs mainly in cases where [error handling](https://nodejs.dev/learn/error-handling-in-nodejs) is not employed. Debuggers can watch behaviors of specific objects throughout program execution by adding breakpoints in a program. Breakpoints are points in your code where you pause execution and examine various factors. By watching objects, you can evaluate their values and type as the execution goes on. 
+
+Although Node comes with a [built-in debugger](https://nodejs.org/api/debugger.html), it's not as flexible. It is a CLI-based debugger where typing things is not an option. Google Chrome DevTools is a non-CLI debugger alternative. It requires no additional installation if you are using a Chromium-based browser like Google Chrome, Chromium and Microsoft Edge, etc. 
+
+## The Setup
+
+Install a Chromium-based browser if you haven't already. You can download Google Chrome [here](https://www.google.com/chrome/).
+
+### On the command line
+
+Run your Node app with the `inspect flag` like this: `node --inspect <filename>`. Example: 
+
+```bash
+$ node --inspect server.js
+```
+ Running your app with the `--inspect` flag allows third-party debuggers to attach to your app and access all the app files. 
+
+You should have an output like this:
+
+```bash
+Debugger listening on ws://127.0.0.1:9229/31dc8b93-3f2c-4352-aa7b-e9357bbabccc
+For help, see: https://nodejs.org/en/docs/inspector
+Listening on http://127.0.0.1:8000
+```
+
+### Debugging a remote app 
+
+If you want to debug your remote in your browser, it's recommended to use SSH tunnels. By doing this, I assume that you understand the security risks associated with allowing remote access to privileged services. Unauthorized database access, website defacement and phishing attacks are some of these risks. 
```

Comment:
If you want to debug your remote in your browser, it's best to use SSH tunnels.

Reply:
However, there are security risks associated with allowing remote access to privileged services. These include unauthorized database access, website defacement and phishing attacks.